src/components/DatePicker/DatePicker.tsx

```diff
@@ -0,0 +1,101 @@
+import { ComponentProps, useEffect, useRef, useState } from 'react';
+import clsx from 'clsx';
+import Calendar, { CalendarProps } from 'react-calendar';
+import { useOnClickOutside } from 'usehooks-ts';
+import { customFormatShortWeekday, unixToDateString } from '@/utils';
+import DateTextField from '../DateTextField/DateTextField';
+import 'react-calendar/dist/Calendar.css';
+import './DatePicker.scss';
+
+interface TDatePickerProps extends ComponentProps<typeof DateTextField> {
+    alignedRight?: boolean;
+    displayFormat?: string;
+    maxDate?: Date;
+    minDate?: Date;
+    mobileAlignment?: 'above' | 'below';
+    onDateChange: (date: string) => void;
+    rightAlignment?: boolean;
+    showLabel?: boolean;
+    value?: string;
+}
+
+const DatePicker = ({
+    alignedRight,
+    disabled,
+    displayFormat = 'MMM DD, YYYY',
+    label,
+    maxDate,
+    minDate,
+    name,
+    onDateChange,
+    rightAlignment = false,
+    showLabel = false,
+    value,
+}: TDatePickerProps) => {
+    const [selectedDate, setSelectedDate] = useState<Date | null>(value ? new Date(value) : null);
+    const [isCalendarOpen, setIsCalendarOpen] = useState(false);
+    const datePickerRef = useRef<HTMLDivElement>(null);
+
+    const toggleCalendar = () => {
+        setIsCalendarOpen(prevState => !prevState);
+    };
+
+    useOnClickOutside(datePickerRef, () => {
+        setIsCalendarOpen(false);
+    });
+
+    const handleDateChange: CalendarProps['onChange'] = value => {
+        const calendarSelectedDate = Array.isArray(value) ? value[0] : value;
+        setSelectedDate(calendarSelectedDate);
+        setIsCalendarOpen(false);
+    };
+
+    useEffect(() => {
+        if (selectedDate !== null) {
+            onDateChange(unixToDateString(selectedDate, displayFormat));
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [selectedDate]);
+
+    const getValue = () => {
+        const isToday = selectedDate !== null && selectedDate.getDate() === new Date().getDate();
+        if (selectedDate !== null && selectedDate !== new Date() && (!isToday || showLabel)) {
+            return unixToDateString(selectedDate, displayFormat);
+        }
+        return '';
```

Comment:
since selectedDate is the main check, we can wrap all of this inside of an if to check for if its not null

```suggestion
        if (selectedData !== null) {
             const isToday = selectedDate.getDate() === new Date().getDate();
             if (selectedDate !== new Date() && (!isToday || showLabel)) {
                  return unixToDateString(selectedDate, displayFormat);
             }
        }
        return '';
```